sat/cli/jobstat/main.py

```diff
@@ -24,49 +24,40 @@
 The main entry point for the jobstat subcommand.
 """
 import logging
-from sat.apiclient.gateway import APIError, APIGatewayClient
 from sat.report import Report
 from sat.config import get_config_value
+from sat.apiclient import APIError
+from sat.apiclient.state_checker import StateCheckerClient
+from sat.session import SATSession
 
 
 LOGGER = logging.getLogger(__name__)
 
 
-def unload_json(url):
-    """Unloads the JSON object from the server url and formats the data as a dictionary"""
-    try:
-        connect = APIGatewayClient()
-        return connect.get(['statechecker', 'jobstat', 'all']).json()
-    except APIError as err:
-        raise APIError(f'Failed to get BOS session templates: {err}')
-    except ValueError as err:
-        raise APIError(f'Failed to parse JSON in response from BOS when '
-                       f'getting session templates: {err}')
-
-
 def do_jobstat(args):
     """Run the jobstat command with the given arguments.
     Args:
         args: The argparse.Namespace object containing the parsed arguments
             passed to this subcommand.
     """
+    state_checker_client = StateCheckerClient(SATSession())
     try:
-        server_data = unload_json(URL)
-        application_data = server_data.get('jobstat')
-        report = Report(
-            tuple(application_data[0].keys()),
-            None,
-            args.sort_by,
-            args.reverse,
-            get_config_value('format.no_headings'),
-            get_config_value('format.no_borders'),
-            filter_strs=args.filter_strs,
-            display_headings=args.fields,
-            print_format=args.format,
-        )
-        for row in application_data:
-            report.add_row(row)
-        print(report)
-    except Exception as err:
-        LOGGER.error(f'Failed to execute jobstat command: {err}')
+        if args.all is True:
```

Comment:
Can this case be removed again? I think in my gist I did not remove it, sorry about that.